python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -806,7 +806,10 @@ def _convert_pydantic_field(pydantic_field: ModelField) -> Field:
 
         return Field(config=wrapped_config_type, description=inferred_field.description)
     else:
-        config_type = _config_type_for_pydantic_field(pydantic_field)
+        if pydantic_field.sub_fields:
+            config_type = _convert_pydantic_field(pydantic_field.sub_fields[0]).config_type
```

Comment:
does this only work if len(pydantic_field.sub_fields) is 1? If not, what happens to sub_fields[1:]?

Reply:
In the case of the Pydantic types we support (list, dict) there is only every a single subfield.

Updated our code to be more explicit about grabbing the subfield from each type.
https://github.com/pydantic/pydantic/blob/v1.10.3/pydantic/fields.py#L758